components/brave_new_tab_ui/components/default/braveToday/customize/Discover.tsx

```diff
@@ -39,48 +39,94 @@ const Hidable = styled.div<{ hidden: boolean }>`
-      .map(p => p.publisherId))
-    ,
-    [lowerQuery, publishers])
+  return <Flex direction='column'>
+    <Header>Discover</Header>
+    <SearchInput type="search" placeholder={getLocale('braveNewsSearchPlaceholderLabel')} value={query} onChange={e => setQuery(e.currentTarget.value)} />
+    { query.length
+      ? <SearchResults query={query} />
+      : <Home />
+    }
+  </Flex>
+}
+
+function Home () {
+  const [showingAllCategories, setShowingAllCategories] = React.useState(false)
+  const channels = useChannels()
+  const { sortedPublishers, filteredPublisherIds } = useBraveNews()
 
   const visibleChannelIds = React.useMemo(() => new Set(channels
-    .filter(c => c.channelName.toLowerCase().includes(lowerQuery))
     // If we're showing all channels, there's no end to the slice.
     // Otherwise, just show the default number.
-    .slice(0, showingAllCategories || query
+    .slice(0, showingAllCategories
       ? undefined
       : DEFAULT_NUM_CATEGORIES)
     .map(c => c.channelName)),
-    [lowerQuery, channels, showingAllCategories])
+    [channels, showingAllCategories])
 
-  return <Flex direction='column'>
-    <Header>Discover</Header>
-    <SearchInput type="search" placeholder={getLocale('braveNewsSearchPlaceholderLabel')} value={query} onChange={e => setQuery(e.currentTarget.value)} />
-    <DirectFeedResults query={query} />
-    <DiscoverSection name={getLocale('braveNewsBrowseByCategoryHeader')}>
+  return (
+    <>
+      <DiscoverSection name={getLocale('braveNewsChannelsHeader')}>
       {channels.map(c => <Hidable key={c.channelName} hidden={!visibleChannelIds.has(c.channelName)}>
         <ChannelCard key={c.channelName} channelId={c.channelName} />
       </Hidable>)}
-      {!showingAllCategories && !query && <LoadMoreButtonContainer>
+      {!showingAllCategories && <LoadMoreButtonContainer>
         <Button onClick={() => setShowingAllCategories(true)}>
-          {getLocale('braveNewsLoadMoreCategoriesButton')}
+            {getLocale('braveNewsLoadMoreCategoriesButton')}
         </Button>
       </LoadMoreButtonContainer>}
-    </DiscoverSection>
-    <DiscoverSection name={getLocale('braveNewsAllSourcesHeader')}>
-      {publishers.map(p => <Hidable key={p.publisherId} hidden={!visiblePublisherIds.has(p.publisherId)}>
-        <FeedCard publisherId={p.publisherId} />
-      </Hidable>)}
-    </DiscoverSection>
-  </Flex>
+      </DiscoverSection>
+      <DiscoverSection name={getLocale('braveNewsAllSourcesHeader')}>
+      {sortedPublishers.map(p =>
+        <Hidable key={p.publisherId} hidden={!filteredPublisherIds.includes(p.publisherId)}>
+          <FeedCard publisherId={p.publisherId} />
+        </Hidable>)}
+      </DiscoverSection>
+    </>
+  )
+}
+
+interface SearchResultsProps {
+  query: string
+}
+function SearchResults (props: SearchResultsProps) {
+  const { sortedPublishers } = useBraveNews()
+  const search = useSearch(props.query)
+  const isFetchable = (search.feedUrlQuery !== null)
```

Comment:
```suggestion
const isFetchable = !!search.feedUrlQuery
```

Reply:
Well, true, but I was explicit about `null` here since `feedUrlQuery` is `string | null`. It would never be an empty string, but it wouldn't hurt to use `!!`.

Reply:
Happy to leave as is.